src/main.rs

```diff
@@ -304,6 +304,12 @@ async fn run_encrypted_payload(
         }
     }
 
+    // run revocation script, if configured
+    let run_revocation = config_get("cloud_agent", "listen_notfications")?;
+    if bool::from_str(&run_revocation.to_lowercase())? {
+        revocation::run_revocation_service();
```

Comment:
I'm not very confident about Rust async stuff, but doesn't this need `.await` as `run_revocation_service()` also returns a `Future`?